server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -248,6 +263,45 @@ public static Metadata upgradeMetadata(Metadata metadata,
         return changed ? upgradedMetadata.build() : metadata;
     }
 
+    // TODO: use one of the upgrader services
+    // bug : https://github.com/crate/crate/issues/13380
+    private static IndexMetadata filterCorruptedIndexMetadata(IndexMetadata indexMetadata, Metadata.Builder upgradedMetadata) {
```

Comment:
When an index is not partitioned (according to the naming) and 1) if contains `partitioned_by` field, try to use this to recover template metadata or 2) if template metadata already exists (which it should not), remove it.